src/spyglass/utils/h5pyfile.py

```diff
@@ -0,0 +1,14 @@
+import os
+
+
+# If SPYGLASS_LINDI is set to 1, use the lindi library to open NWB files.
+# Otherwise, use h5py. Note that even in LINDI mode, if the file path does not
+# end with ".lindi.json", then H5pyFile is equivalent to h5py.File. Thus, the
+# difference only matters when opening LINDI files.
+
+USE_LINDI = os.environ.get("SPYGLASS_LINDI", "0") == "1"
+
+if USE_LINDI:
+    from lindi import File as H5pyFile  # noqa: F401
+else:
+    from h5py import File as H5pyFile  # noqa: F401
```

Comment:
It seems like there are quite a few places where you're using multiple contexts
```python
with H5pyFile(analysis_nwb_file_abs_path, "a") as h5f:
        with pynwb.NWBHDF5IO(
            file=h5f,
            mode="a",
            load_namespaces=True,
        ) as io:
```

Is it possible to define a context manager here that would clean up those instances by combining them into something like ...?

```python
with H5pyNwb(
    analysis_nwb_file_abs_path,
    mode="a",
    load_namespaces=True,
) as h5f:
```

Reply:
Yeah I that would be possible, but I thought it's better to be explicit.

Reply:
Sure, I think I was part motivated by the readability of the diff, and part motivated by a desire to see spyglass have less nesting than it does. If it's sometimes just one or the other, we can keep them separate. If it's always both, I'd be in favor of combining